src/values/AbstractValue.js

```diff
@@ -245,31 +245,31 @@ export default class AbstractValue extends Value {
   mightBeNull(): boolean {
     let valueType = this.getType();
     if (valueType === NullValue) return true;
-    if (valueType !== Value) return false;
+    if (valueType !== PrimitiveValue && valueType !== Value) return false;
```

Comment:
Until widening came in to the picture, no test case caused an abstract value with type === PrimitiveValue.